Algorithm/QCAlgorithm.Indicators.cs

```diff
@@ -2530,6 +2530,75 @@ private void WarmUpIndicatorImpl<T>(Symbol symbol, TimeSpan period, Action<T> ha
 
             SubscriptionManager.RemoveConsolidator(symbol, consolidator);
         }
+        
+        /// <summary>
+        /// Warms up a given list of indicators with historical data
+        /// </summary>
+        /// <param name="symbol">The symbol whose indicators we want</param>
+        /// <param name="indicators">The indicators we want to warm up</param>
+        /// <param name="period">The necessary period to warm up the indicators</param>
```

Comment:
In some cases, the minimum number of samples that an indicator needs to be fully warmed up is not trivial, so the user might provide a period that is not big enough. Consequently, the `period` should be defined by the maximum value of `IIndicatorWarmUpPeriodProvider.WarmUpPeriod` (see `GetIndicatorWarmUpHistory` for reference), and this parameter can be removed.

Reply:
I've been thinking about that, too. But what about custom python indicators which don't have a `WarmUpPeriod` as far as I know? Maybe a better solution would be setting `int period = null` as default and if period is null, we use the approach you described. What do you think?